google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -81,6 +81,7 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import javax.activation.MimetypesFileTypeMap;
```

Comment:
Use com.google.common.net.MediaType so you don't introduce another dependency, and ping the guava team to get this out of beta

Reply:
com.google.common.net.MediaType doesn't provide mapping extension to the meida type. Its parse method recognizes strings like "text/plain".

Reply:
What about Files.probeContentType(Path)?

Reply:
or URLConnection

Reply:
Both Files.probeContentType(Path) and URLConnection work! I will use it instead. Thanks for the hint.